Sources/SwiftDiagnostics/GroupedDiagnostics.swift

```diff
@@ -275,9 +293,25 @@ extension GroupedDiagnostics {
 extension DiagnosticsFormatter {
   /// Annotate all of the source files in the given set of grouped diagnostics.
   public func annotateSources(in group: GroupedDiagnostics) -> String {
-    return group.rootSourceFiles.map { rootSourceFileID in
-      group.annotateSource(rootSourceFileID, formatter: self, indentString: "")
-    }.joined(separator: "\n")
+    // Render all grouped diagnostics.
+    var renderedDiags = group.rootSourceFiles.compactMap { rootSourceFileID in
+      let annotated = group.annotateSource(
+        rootSourceFileID,
+        formatter: self,
+        indentString: ""
+      )
+      return annotated.isEmpty ? nil : annotated
+    }
+
+    // Render any floating diagnostics, which have no anchor.
+    renderedDiags.append(
```

Comment:
Opinionated question: Should we render the floating diagnostics before or after the diagnostics with source locations? I would have rendered them before because I visualize them on the first line.

Also, using `+=` instead of `append(contentsOf:)` would remove a nesting level here.